automatically set ELB_LIST if not provided

wjordan opened the pull request:
This PR removes the need to manually set `ELB_LIST` when not using an auto-balancer, and removes the step in the README.md where this is required.

In `deregister_from_elb.sh`, if `ELB_LIST` is empty this PR calls the existing `get_elb_list` function to automatically find all load balancers the instance is currently registered to, then writes this list to a temp file (`/tmp/elblist`).

In `register_with_elb.sh`, if `ELB_LIST` is empty this PR reads the list of previously-registered load balancers from the temp file (`/tmp/elblist`).

The behavior is unchanged when an auto-scaling group is used.

yubangxi commented:
Hi,

Thanks for the PR. 

Although it might not be the best practice, there could be multiple applications running on the same instance but behind different ELB at the same time. It could be error-prone to automatically deregister and register instance from all ELBs it's bund to. I think it would be helpful if we provide this feature as an opt-in option instead of default behavior.

wjordan commented:
I've updated the PR to make this feature opt-in.
Also, I found that this feature requires a fix to the `get_elb_list` function, which I've added in a separate PR #41.

suryanarayanan commented:
Could you please confirm this contribution is under the terms of the Apache 2.0 license. Thanks.

wjordan commented:
Confirmed, Apache 2.0 license applies to this contribution.
